src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.Resolution.cs

```diff
@@ -772,13 +774,13 @@ private static void AddModule(PEModule module, int referenceIndex, ResolvedRefer
             TCompilation compilation,
             DiagnosticBag diagnostics,
             out ImmutableArray<MetadataReference> references,
-            out IDictionary<(string, string), MetadataReference> boundReferenceDirectives,
+            [NotNull] out IDictionary<(string, string), MetadataReference>? boundReferenceDirectives,
```

Comment:
The `?` annotation here seems incorrect, as it exposes an internal implementation detail from this method. It doesn't matter very much here since it's not a public API, but I would recommend using a temp instead.